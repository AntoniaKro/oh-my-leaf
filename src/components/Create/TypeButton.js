import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const StyledInputButton = styled.input`
  background-color: ${props => (props.filling ? '#ffd084' : 'transparent')};
  border: 2px solid #ffd084;
  border-radius: 11px;
  font-family: 'PT Mono', monospace;
  font-size: 18px;
  margin: 8px;
  padding: 5px;
`;

function TypeButton(props) {
  const { onClick, value, filled } = props;
  return (
    <StyledInputButton
      onClick={onClick}
      type="button"
      value={value}
      filling={value === filled}
    />
  );
}

TypeButton.propTypes = {
  handleClick: PropTypes.func,
  value: PropTypes.string,
  filled: PropTypes.string
};

export default TypeButton;
