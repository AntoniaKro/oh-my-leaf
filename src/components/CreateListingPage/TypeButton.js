import React from 'react';
import styled from 'styled-components';

const StyledInputButton = styled.input`
  border: 2px solid #ffd084;
  margin: 8px;
  border-radius: 11px;
  font-size: 18px;
  padding: 5px;
  font-family: 'PT Mono', monospace;
  background-color: ${props => (props.filling ? '#ffd084' : 'transparent')};
`;

function TypeButton(props) {
  const { handleClick, value, filled } = props;
  return (
    <StyledInputButton
      onClick={handleClick}
      type="button"
      value={value}
      filling={value === filled}
    />
  );
}

export default TypeButton;
