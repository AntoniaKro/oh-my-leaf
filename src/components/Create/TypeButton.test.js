import React from 'react';
import TypeButton from './TypeButton';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const types = ['give away', 'swap', 'for sale'];

Enzyme.configure({ adapter: new Adapter() });

function value() {
  return types[0];
}

describe('Type button component', () => {
  it('renders a button in the form component to choose the type of a listing', () => {
    const button = renderer.create(<TypeButton value={value()} />);
    const tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('triggers the callback and updates the type state in Create Form when clicked', () => {
    const callback = jest.fn();
    const button = mount(<TypeButton onClick={callback} value={value()} />);
    button.simulate('click');
    expect(callback).toHaveBeenCalled();
  });
});
