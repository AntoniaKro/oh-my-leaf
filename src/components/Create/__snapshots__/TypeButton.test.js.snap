// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type button component renders a button in the form component to choose the type of a listing 1`] = `
<input
  className="TypeButton__StyledInputButton-p6v7sf-0 izjMxW"
  type="button"
  value="give away"
/>
`;
